chia/simulator/full_node_simulator.py

```diff
@@ -123,7 +123,34 @@ async def update_autofarm_config(self, enable_autofarm: bool) -> bool:
             return self.auto_farm
 
     async def get_all_coins(self, request: GetAllCoinsProtocol) -> list[CoinRecord]:
-        return await self.full_node.coin_store.get_all_coins(request.include_spent_coins)
```

Comment:
Having this method in `CoinStoreProtocol` is a bad idea. We use an equivalent algorithm, also a bad idea, but at least contained to the simulator, limiting its potential damage.

Reply:
Good idea.